pkg/apis/operator/v1alpha1/tektonaddon_types.go

```diff
@@ -65,6 +67,17 @@ type TektonAddonStatus struct {
 	AddonsInstallerSet map[string]string `json:"installerSets,omitempty"`
 }
 
+// Addon defines the field to customize Addon component
+type Addon struct {
+	// Params is the list of params passed for Addon customization
+	// +optional
+	Params []Param `json:"params,omitempty"`
+}
+
+func (a Addon) IsEmpty() bool {
+	return reflect.DeepEqual(a, Addon{})
```

Comment:
Any reason to use `reflect` here instead of just `len(a.Params) == 0` ?
We should avoid using `reflect` when we can, it's kind-of costly for.. not always a valid reason to be.

Reply:
yeah len of param would work... 
@vdemeester how about using `equality.Semantic.DeepEqual` not in this case but [here](https://github.com/tektoncd/operator/blob/039009587fe6bd89f975455a98c39113530f097e/pkg/reconciler/shared/tektonconfig/pipeline/pipeline.go#L61) is that okay compared to reflect?

Reply:
Maybe yes ? My point is just that we should aim to use as less as possible `reflect` and those things.. but when we don't have the choice .. 😛